pkg/build/raw.go

```diff
@@ -86,12 +86,14 @@ func (b *Builder) writeModifyScript(imageFilename string, includeCombustion, ren
 		ConfigureGRUB       string
 		ConfigureCombustion bool
 		RenameFilesystem    bool
+		DiskSize            string
 	}{
 		ImagePath:           imageFilename,
 		CombustionDir:       b.context.CombustionDir,
 		ConfigureGRUB:       grubConfiguration,
 		ConfigureCombustion: includeCombustion,
 		RenameFilesystem:    renameFilesystem,
+		DiskSize:            b.context.ImageDefinition.OperatingSystem.RawConfiguration.DiskSize,
```

Comment:
It may work, but the validation I've added makes sure the user specifies a capitalised M/G/T, that way we know that it will input into the command properly.